geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/MessageDispatcher.java

```diff
@@ -362,31 +385,63 @@ protected void runDispatcher() {
           }
           waitForResumption();
         }
-        try {
-          clientMessage = (ClientMessage) _messageQueue.peek();
-        } catch (RegionDestroyedException skipped) {
-          break;
+
+        if (clientMessage == null) {
+          try {
+            clientMessage = (ClientMessage) _messageQueue.peek();
+          } catch (RegionDestroyedException skipped) {
+            break;
+          }
         }
+
         getStatistics().setQueueSize(_messageQueue.size());
         if (isStopped()) {
           break;
         }
-        if (clientMessage != null) {
-          // Process the message
-          long start = getStatistics().startTime();
-          //// BUGFIX for BUG#38206 and BUG#37791
-          boolean isDispatched = dispatchMessage(clientMessage);
-          getStatistics().endMessage(start);
-          if (isDispatched) {
+
+        if (clientMessage == null) {
+          _messageQueue.remove();
+          break;
+        }
+
+        // Process the message
+        long start = getStatistics().startTime();
+        try {
+          if (dispatchMessage(clientMessage)) {
+            getStatistics().endMessage(start);
             _messageQueue.remove();
             if (clientMessage instanceof ClientMarkerMessageImpl) {
               getProxy().setMarkerEnqueued(false);
             }
           }
-        } else {
+          clientMessage = null;
+          wait_for_re_auth_start_time = -1;
+        } catch (NotAuthorizedException notAuthorized) {
+          // behave as if the message is dispatched, remove from the queue
+          logger.info("skip delivering message: " + clientMessage, notAuthorized);
```

Comment:
this exception should not happen in the first place, if it does, an info level message would make it clear what happened since this exception is not bubbled up to the caller.

Reply:
But if it does happen (i.e. the info log message) do we want the server to keep logging it for every message it attempts to send to that unauthenticated client? Will the server close this dispatcher in this case or retry reauth? If not it seems like we could log once (maybe it should be a warning) for that client.

Reply:
Jinmei, Darrel has additional comment on this...I am changing the state to unresolved.

Reply:
Notice this is NotAuthorizedException (not AuthenticationExpiredException), when this exception happens, we only log it, drop it from the queue and continue on to the next message (that's what the old authorization model does to the un-authorized messages), we don't do retrys for any "un-authorized" messages.